src/net.cpp

```diff
@@ -1868,6 +1889,11 @@ bool CConnman::AddConnection(const std::string& address, ConnectionType conn_typ
 
 void CConnman::DisconnectNodes()
 {
+    AssertLockNotHeld(m_nodes_mutex);
+    AssertLockNotHeld(m_reconnections_mutex);
+
+    decltype(m_reconnections) reconnections_to_add;
```

Comment:
6a2dbed1209b18d923da64b019f6f378d3c30695: could a comment here that we `// Use temporary variable to track new reconnection candidates, because m_reconnections_mutex may be blocked [also for several seconds??? or much shorter but still don't want that]` (supplementing the comment below why we don't do the reconnect itself here either).